FenwickTree/RangeUpdatePointQueries.java

```diff
@@ -0,0 +1,87 @@
+public class RangeUpdatePointQueries {
+	
+	public static void main(String[] args) {
+		RangeUpdatePointQueries d=new RangeUpdatePointQueries();
+		//input array
+		int[] inArr={0,0,0,0,0,0};
+		//Store the fenwick tree
+		int[] feniTree=new int[inArr.length+1];
+		//Update range using method 2
+		d.updateRangeMeth2(inArr, 2, 4, 10);
```

Comment:
What is the difference between updateRange and updateRangeMeth2? When should one be used over the other? Furthermore, why is _inArr_ being passed into the method and not _feniTree_?

Reply:
updateRangeMeth2 updates the input array(it has linear complexity) and doesn't involves the fenwick tree, hence it is getting passed in the method. I will remove the linear time method, refactor the code and push it again in the branch.